gpkit/solution_array.py

```diff
@@ -592,6 +593,25 @@ def savetxt(self, filename="solution.txt", *, printmodel=True, **kwargs):
                 f.write(self.modelstr + "\n")
             f.write(self.table(**kwargs))
 
+    def savejson(self, filename="solution.json"):
+        "Saves solution table as a json file"
+        sol_dict = {}
+        # add appropriate data for each variable to the dictionary
+        for key in self.name_collision_varkeys():
+            key.descr["necessarylineage"] = True
+        data = self["variables"]
+        for i, (k, v) in enumerate(data.items()):
+            key = str(k)
+            if isinstance(v, np.ndarray):
+                val = {'v': v.tolist(), 'u': str(k.unitstr())}
+            else:
+                val = {'v': v, 'u': str(k.unitstr())}
+            sol_dict[key] = val
+        for key in self.name_collision_varkeys():
+            del key.descr["necessarylineage"]
+        with open(filename, "w") as f:
+            json.dump(sol_dict, f)
```

Comment:
should this also return the json as a string, in case the user wants it? (i.e. `return json.dumps(sol_dict)`)

maybe if the filename is `None` it doesn't save and only does that.

Reply:
created a boolean called printjson that will return the string if flagged true

Reply:
hmm maybe `returnjson`?